helix-term/src/keymap.rs

```diff
@@ -303,6 +303,14 @@ impl Keymaps {
         self.sticky.as_ref()
     }
 
+    pub fn key_exists(&self, mode: Mode, key: KeyEvent) -> bool {
+        let keymaps = &*self.map();
+        let keymap = &keymaps[&mode];
+        let mut ks = self.pending().to_vec();
+        ks.push(key);
+        keymap.search(&ks).is_some()
```

Comment:
We can avoid the allocation of another vec for the pending keys here:

```suggestion
        keymap
            .search(self.pending())
            .and_then(KeyTrie::node)
            .is_some_and(|node| node.contains_key(&key))
```